src/WingetCreateCore/Common/PackageParser.cs

```diff
@@ -1116,5 +1119,23 @@ private static string RemoveInvalidCharsFromString(string value)
         {
             return Regex.Replace(value, InvalidCharacters, string.Empty);
         }
+
+        private static void ValidateUrl(string url, bool allowHttp)

```

Comment:
Since you've added an explicit function for checking if a string is a valid URI for our needs, we _may_ want to change this code in UpdateCommand.cs (that checks if the param `--release-notes-url` is a valid URL) to re-use this function for consistency

https://github.com/microsoft/winget-create/blob/c92436b01095e45b054652116b7ff0eef28c7ca0/src/WingetCreateCLI/Commands/UpdateCommand.cs#L170-L178

Reply:
This change specifically targets scenarios where bits are downloaded. We retrieve the installer from the provided input URL to parse metadata such as the publisher, version, and other relevant details. To enhance security, we want to ensure that both the download and the parsing of the package occur over a secure protocol, such as HTTPS.